crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3521,21 +3530,17 @@ impl<'db> Type<'db> {
         self,
         db: &'db dyn Db,
         name: &str,
-        mut argument_types: CallArgumentTypes<'_, 'db>,
+        argument_types: &mut CallArgumentTypes<'_, 'db>,
         policy: MemberLookupPolicy,
     ) -> Result<Bindings<'db>, CallDunderError<'db>> {
         match self
-            .member_lookup_with_policy(
-                db,
-                name.into(),
-                MemberLookupPolicy::NO_INSTANCE_FALLBACK | policy,
```

Comment:
Callers must now provide the full policy that they want to use, and `NO_INSTANCE_FALLBACK` is not implicitly added. That allows us to use `try_call_dunder_with_policy` (instead of copy/pasting it) down below in `try_call_constructor`, to find and call the `__new__` class method.